src/main/java/com/mael/ttt/GameSetup.java

```diff
@@ -0,0 +1,30 @@
+package com.mael.ttt;
+
+import com.mael.ttt.players.Player;
+import com.mael.ttt.ui.UserInterface;
+import com.mael.ttt.ui.menu.MenuCreator;
+
+import java.util.List;
+
+public class GameSetup {
+    private UserInterface gameUI;
+
+    public GameSetup(UserInterface gameUI) {
+        this.gameUI = gameUI;
+    }
+
+    public void setUp() {
+        gameUI.printWelcomeMessage();
+    }
+
+    public void playGame() {
+        do {
+            Game game = new Game(new Board(3), gameUI, setPlayers());
+            game.start();
+        } while (gameUI.replay().equals("y"));
+    }
+
+    private List<Player> setPlayers() {
+        return new MenuCreator().createMenu(gameUI).createPlayers();
```

Comment:
Is it worth passing the gameUI through the constructor instead of a `createMenu(...)` method? I personally use  constructor arguments when at point in the code where I create the instance I already know the value of the arguments. I use `create...()` methods when the value that gets passed depends on user input or something I can't pull all the way up to where I create the instance.